sqlx-sqlite/src/connection/mod.rs

```diff
@@ -426,3 +426,8 @@ impl Statements {
         self.temp = None;
     }
 }
+
+pub(crate) enum Returning {
+    Many,
+    One,
+}
```

Comment:
You don't need a bespoke enum. I would just take a `limit: Option<usize>` parameter and stop reading rows when the limit is reached. For `.fetch_optional()`, you'd pass `Some(1)`.

Reply:
@abonander good call, updated.